### ISSUE TITLE
Crontab tries to upload delivery report before analysis is done

### ISSUE THREAD

**Issue by moahaegglund** [Clinical-Genomics/cg#1127]
## Description

Case helpedpossum was analysed as a single case on the 9th of april and the delivery report was uploaded that day. Customer ordered rerun of this case and added the parents to it on the 20th of april. The crontab for uploading the delivery report tries to upload the delivery report for the trio since yesterday. This doesn't work since the analysis doesn't exist yet.

```
2021-04-21 12:21:58 hasta.scilifelab.se cg.cli.upload.delivery_report[74718] INFO Uploading delivery report for case: helpedpossum
----------------- DELIVERY_REPORT -------------
2021-04-21 12:21:58 hasta.scilifelab.se cg.cli.upload.delivery_report[74718] INFO Using analysis started at: 2021-04-08 13:56:35
2021-04-21 12:22:00 hasta.scilifelab.se cg.utils.commands[74718] INFO Running command /home/proj/bin/conda/envs/P_scout/bin/scout --config /home/proj/production/servers/config/hasta.scilifelab.se/scout.yaml export panel IEM
2021-04-21 12:22:20 hasta.scilifelab.se cg.utils.commands[74718] INFO Running command /home/proj/bin/conda/envs/P_scout/bin/scout --config /home/proj/production/servers/config/hasta.scilifelab.se/scout.yaml export panel mtDNA
2021-04-21 12:22:38 hasta.scilifelab.se cg.utils.commands[74718] INFO Running command /home/proj/bin/conda/envs/P_chanjo/bin/chanjo --config /home/proj/stage/servers/config/hasta.scilifelab.se/chanjo.yaml calculate coverage -s ACC7642A34 -f /home/hiseq.clinical/tmp/tmp5bgfzg4u
2021-04-21 12:22:46 hasta.scilifelab.se cg.utils.commands[74718] INFO Running command /home/proj/bin/conda/envs/P_chanjo/bin/chanjo --config /home/proj/stage/servers/config/hasta.scilifelab.se/chanjo.yaml calculate coverage -s ACC7723A1 -f /home/hiseq.clinical/tmp/tmpy9shgmch
2021-04-21 12:22:47 hasta.scilifelab.se cg.utils.commands[74718] INFO Running command /home/proj/bin/conda/envs/P_chanjo/bin/chanjo --config /home/proj/stage/servers/config/hasta.scilifelab.se/chanjo.yaml calculate coverage -s ACC7723A2 -f /home/hiseq.clinical/tmp/tmp9thl7exb
2021-04-21 12:22:47 hasta.scilifelab.se cg.apps.housekeeper.hk[74718] INFO Fetch latest version from bundle helpedpossum
2021-04-21 12:22:47 hasta.scilifelab.se housekeeper.store.api.find[74718] INFO Fetching files from bundle helpedpossum
2021-04-21 12:22:47 hasta.scilifelab.se housekeeper.store.api.find[74718] INFO Fetching files with tags in [mip-config]
2021-04-21 12:22:47 hasta.scilifelab.se housekeeper.store.api.find[74718] INFO Fetching files from version 35407
2021-04-21 12:22:48 hasta.scilifelab.se cg.apps.housekeeper.hk[74718] INFO Fetch latest version from bundle helpedpossum
2021-04-21 12:22:48 hasta.scilifelab.se housekeeper.store.api.find[74718] INFO Fetching files from bundle helpedpossum
2021-04-21 12:22:48 hasta.scilifelab.se housekeeper.store.api.find[74718] INFO Fetching files with tags in [qcmetrics]
2021-04-21 12:22:48 hasta.scilifelab.se housekeeper.store.api.find[74718] INFO Fetching files from version 35407
2021-04-21 12:22:48 hasta.scilifelab.se cg.apps.housekeeper.hk[74718] INFO Fetch latest version from bundle helpedpossum
2021-04-21 12:22:48 hasta.scilifelab.se housekeeper.store.api.find[74718] INFO Fetching files from bundle helpedpossum
2021-04-21 12:22:48 hasta.scilifelab.se housekeeper.store.api.find[74718] INFO Fetching files with tags in [sampleinfo]
2021-04-21 12:22:48 hasta.scilifelab.se housekeeper.store.api.find[74718] INFO Fetching files from version 35407
2021-04-21 12:22:49 hasta.scilifelab.se cg.cli.upload.delivery_report[74718] ERROR Creation of delivery report failed for case: helpedpossum, Could not generate report data for helpedpossum, missing data: ['analysis_sex', 'mapped_reads', 'duplicates', 'mapped_reads', 'duplicates', 'analysis_sex', 'analysis_sex', 'mapped_reads', 'duplicates', 'mapped_reads', 'duplicates', 'analysis_sex']
```

**Comment by karlnyr** [comment]
I think this is arguable if its a bug in delivery reports or order portal. Since one should not be able to add old/new samples to an existing case. @moahaegglund did you link the family manually or was the customer able to place an order with these three samples?

**Comment by moahaegglund** [comment]
No, I only started the analysis, didn't link anything. The customer was able to place an order with these samples.

**Comment by karlnyr** [comment]
To debrief, this case the customer probably by mistake placed an order for a singleton out of a trio case. At the time of the order it was not possible to do so but a fix was implemented in https://github.com/Clinical-Genomics/cg/pull/1095. This fix allowed the customer to order re-runs of a singleton by which it was analysed as one. The customer probably then realised that this was not their intention and ordered the case again as a trio case. 

I think we should restrict the addition of old samples to old cases (similar to new samples to old cases were a new case has to be created) @patrikgrenfeldt I believe we allow customers to add old samples to old families, or am I wrong? We could make them enter a new family name for a new family constellation.

**Comment by karlnyr** [comment]
From discussions with @patrikgrenfeldt:
Cases are always treated as new ones in the back-end of the orderportal, we need to ensure wether or not a case is existing or not for the customer, and if it is we won’t add any samples to it but request the user to enter a new family name. It seems that frontend is keeping information in the back-end about an internal id for cases, but it is not communicating this to the back-end.

**Comment by patrikgrenfeldt** [comment]
closing due to fix #1154
